lib/posix.dart

```diff
@@ -31,8 +35,33 @@ class _PosixMutex extends Mutex {
         super._();
 
   @override
-  void _lock() {
-    if (pthread_mutex_lock(_impl) != 0) {
+  void _lock({Duration? timeout}) {
+    if (timeout == null) {
+      if (pthread_mutex_lock(_impl) != 0) {
+        throw StateError('Failed to lock mutex');
+      }
+    } else {
+      _timedLock(timeout);
+    }
+  }
+
+  void _timedLock(Duration timeout) {
+    final timespec =
+        malloc.allocate<pthread_timespec_t>(sizeOf<pthread_timespec_t>());
+
+    /// calculate the absolute timeout in seconds
```

Comment:
nit: `// Calculate ...`

nit: `abstimeout` is better called something like `wakeupTime` or something like that.

Also seems strange to just discard milliseconds given that underlying OS supports timeouts with larger precision.

Reply:
done.